src/sync/condvar.rs

```diff
@@ -138,15 +141,16 @@ impl Condvar {
         trace!(waiters=?state.waiters, next_epoch=state.next_epoch, "woken from condvar {:p}", self);
         let my_status = state.waiters.remove(&me).expect("should be waiting");
         match my_status {
-            CondvarWaitStatus::Broadcast => {
-                // Woken by a broadcast, so nothing to do
+            CondvarWaitStatus::Broadcast(clock) => {
+                // Woken by a broadcast, so nothing to do except update the clock
+                ExecutionState::with(|s| s.update_clock(&clock));
             }
             CondvarWaitStatus::Signal(mut epochs) => {
-                let epoch = epochs.pop_front().expect("should be a pending signal");
+                let (epoch, clock) = epochs.pop_front().expect("should be a pending signal");
```

Comment:
Maybe we can think about this in a future PR, but: does having vector clocks subsume the need for epochs? We just want to know whether a thread can "see" a notify, and it seems like that's exactly what causality would tell us?

Reply:
I added a TODO to take a look.  We can also create an issue if we want to track it better.

Reply:
Created #39 if you want to link it from the todo.

Reply:
Done - thanks!